### PR TITLE
Added GetProfile Controller

### LINKED ISSUE #45: Get your own profile
@route GET api/profile/me 
@desc Get your own user profile 
@access Private

### PR DESCRIPTION
**Pull request by Shubham-1105**
resolves #45

### CODE REVIEW

**Comment by BhavyaaArora-08** [comment]
Can you please attach screenshots as well.
You can use Postman to test the route.

**Comment by Shubham-1105** [comment]
actually the post route is not done for storing user profile details in db.
Is the code ok,we will get the user id from request?

File: server/src/routers/api/userProfileRouter.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// @access Private
+
+router.get("/me", async (req, res) => {
+  try {
+    const userProfile = await UserProfile.findOne({
+      _id: req.params.userProfileId,
+    });
+    if (!userProfile) {
+      return res.status(400).json({
+        status: "fail",
```
Comment (BhavyaaArora-08) [inline server/src/routers/api/userProfileRouter.js:17]: please refer the response format and adhere to the names

File: server/src/routers/api/userProfileRouter.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        message: "No user found",
+      });
+    }
+
+    res.status(200).json({
+      status: "success",
+      message: "user profile found",
+      data: userProfile,
```
Comment (BhavyaaArora-08) [inline server/src/routers/api/userProfileRouter.js:25]: there is a proper response format, dm me on slack if you don't have that.

File: server/src/routers/api/userProfileRouter.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+const router = express.Router();
+const UserProfile = require("../../models/Profile");
+
+// @route GET api/profile/me
+// @desc Get your own user profile
+// @access Private
+
+router.get("/me", async (req, res) => {
+  try {
+    const userProfile = await UserProfile.findOne({
```
Comment (BhavyaaArora-08) [inline server/src/routers/api/userProfileRouter.js:12]: This is a private route so include auth middleware,
router.get("/me", auth, async(req,res)=>{
// here user who is trying to access this route is available at req.user now ( this was done in the auth middleware have a look at it)
});

**Comment by BhavyaaArora-08** [comment]
Also, wait for the post profile route to be submitted, then you can include the screenshots.
I won't unassign the issue to you, so don't worry about that.

**Comment by BhavyaaArora-08** [comment]
Are you working on the issue?

**Comment by BhavyaaArora-08** [comment]
Please add screenshots.
You can use Postman to test the API endpoints.

**Pull request closed without merging**